ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -837,7 +900,52 @@ int fillFrom(ByteBuffer channelReadBuffer) {
         buf.put(channelReadBuffer);
       }
       if (!buf.hasRemaining()) {
-        onFillComplete(true);
+        if (!passedInBlobProperties.isEncrypted()) {
+          buf.flip();
+          onFillComplete(true);
+          updateChunkFillerWaitTimeMetrics();
+        } else {
+          try {
+            updateChunkFillerWaitTimeMetrics();
+            this.state = ChunkState.Encrypting;
+            logger.trace("Chunk state moving to {} for chunk at index {}", ChunkState.Encrypting, chunkIndex);
+            buf.flip();
+            chunkBlobSize = buf.remaining();
+            logger.trace("Submitting encrypt job for chunk at index {}", chunkIndex);
+            exec.submitJob(
```

Comment:
Why is this getting duplicated in `fillFrom()` and `fillChunks()`? Is this about the last chunk? You should really move this to a common method.

Reply:
Sure. Will try to see if I can club it as one.